acmw-db-app/pages/reset.js

```diff
@@ -0,0 +1,176 @@
+    let input = this.state.input;
+    input[event.target.name] = event.target.value;
+  
+    this.setState({
+      input
+    });
+  }
+     
+  handleSubmit(event) {
+    event.preventDefault();
+  
+    if(this.validate()){
+        console.log(this.state);
+  
+        let input = {};
+        input["name"] = "";
+        input["email"] = "";
+        input["password"] = "";
+        input["confirm_password"] = "";
+        this.setState({input:input});
+  
+        alert('Passwords match. Form "submitted"');
+    }
+  }
+  
+  validate(){
+      let input = this.state.input;
+      let errors = {};
+      let isValid = true;
+  
+      if (!input["name"]) {
+        isValid = false;
+        errors["name"] = "Please enter your name.";
+      }
+  
+      if (!input["email"]) {
+        isValid = false;
+        errors["email"] = "Please enter your email Address.";
+      }
+  
+      if (typeof input["email"] !== "undefined") {
+          
+        var pattern = new RegExp(/^(("[\w-\s]+")|([\w-]+(?:\.[\w-]+)*)|("[\w-\s]+")([\w-]+(?:\.[\w-]+)*))(@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$)|(@\[?((25[0-5]\.|2[0-4][0-9]\.|1[0-9]{2}\.|[0-9]{1,2}\.))((25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\.){2}(25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\]?$)/i);
+        if (!pattern.test(input["email"])) {
+          isValid = false;
+          errors["email"] = "Please enter valid email address.";
+        }
+      }
+  
+      <h6>Please enter a password containing at least 1 uppercase, 1 lowercase, 1 digit, 1 special character with a length of at least of 8</h6>
+      var pattern = new RegExp(/^(?=.{8,}$)(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*\W).*$/i);
+      if (!pattern.test(input["password"])) {
+        isValid = false;
+        errors["password"] = "Please try a new password with the given constraints: 1 uppercase, 1 lowercase, 1 special character, length of 8+";
+      }
+      if (!input["password"]) {
+        isValid = false;
+        errors["password"] = "Please enter your password.";
+      }
+  
+      if (!input["confirm_password"]) {
+        isValid = false;
+        errors["confirm_password"] = "Please confirm your password.";
+      }
+  
+      if (typeof input["password"] !== "undefined" && typeof input["confirm_password"] !== "undefined") {
+          
+        if (input["password"] != input["confirm_password"]) {
+          isValid = false;
+          errors["password"] = "Passwords don't match.";
+        }
+      } 
+  
+      this.setState({
+        errors: errors
+      });
+  
+      return isValid;
+  }
+     
+  render() {
+    return (
+      <div className={styles.container}>
+        <Head>
+        <title>ACM-W Database Password Reset</title>
+        </Head>
+
+      <main className={styles.card}>
+        <h1>Password Reset</h1>
+        <form onSubmit={this.handleSubmit}>
+  
+          <div className={styles.passwordChunk}>
+            <label for="name">Name:</label>
+            <input 
```

Comment:
I got these two errors in console
```
Warning: Invalid DOM property `for`. Did you mean `htmlFor`?
    in label (at reset.js:110)
    in div (at reset.js:109)
    in form (at reset.js:107)
    in main (at reset.js:105)
    in div (at reset.js:100)
    in ResetPW (at _app.js:4)
    in MyApp
    in Unknown
    in Context.Provider
    in Context.Provider
    in Context.Provider
    in Context.Provider
    in AppContainer
Warning: Invalid DOM property `class`. Did you mean `className`?
    in input (at reset.js:111)
    in div (at reset.js:109)
    in form (at reset.js:107)
    in main (at reset.js:105)
    in div (at reset.js:100)
    in ResetPW (at _app.js:4)
    in MyApp
    in Unknown
    in Context.Provider
    in Context.Provider
    in Context.Provider
    in Context.Provider
    in AppContainer
```

Reply:
Got these errors as well, to fix, you can do:
`<label htmlFor="name">`
and then just delete the "class" attribute in input (I don't think you have one you're using)